agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -2210,21 +2210,26 @@ bool backhaul_manager::handle_1905_topology_query(ieee1905_1::CmduMessageRx &cmd
      * Set the number of local interfaces and fill info of each of the local interfaces, according
      * to IEEE_1905 section 6.4.5
      */
-    uint32_t speed;
-    if (network_utils::linux_iface_get_speed(m_sConfig.wire_iface, speed)) {
-        std::shared_ptr<ieee1905_1::cLocalInterfaceInfo> localInterfaceInfo =
-            tlvDeviceInformation->create_local_interface_list();
 
+    /**
+     * Add a LocalInterfaceInfo field for the wired interface, if any.
+     */
+    std::string local_interface_name = m_sConfig.wire_iface;
+    // TODO: check if interface is UP?
```

Comment:
Why not check it then?
For example:
```
 // If a wired (WAN) interface was provided, try it first, check if the interface is UP
            wan_monitor::ELinkState wired_link_state = wan_monitor::ELinkState::eInvalid;
            if (!m_sConfig.wire_iface.empty()) {
                wired_link_state = wan_mon.initialize(m_sConfig.wire_iface);
                // Failure might be due to insufficient permissions, datailed error message is being
                // printed inside.
                if (wired_link_state == wan_monitor::ELinkState::eInvalid) {
                    LOG(WARNING) << "wan_mon.initialize() failed, skip wired link establishment";
                }
            }
            if ((wired_link_state == wan_monitor::ELinkState::eUp) &&
                (selected_backhaul.empty() || selected_backhaul == DEV_SET_ETH)) {
```

Reply:
Done. 
However, I've used `network_utils::linux_iface_is_up_and_running()` to check interface status because it's simpler than your proposal and has no side effects.